src/transformers/modeling_attn_mask_utils.py

```diff
@@ -66,7 +66,7 @@ def to_causal_4d(
         batch_size: int,
         query_length: int,
         key_value_length: int,
-        dtype: torch.dtype = torch.float32,
+        dtype: torch.dtype,
```

Comment:
It is fine to remove the default (that is the cause of the error), as in `_prepare_4d_causal_attention_mask`, `_prepare_4d_attention_mask` and `_create_4d_causal_attention_mask` privately exposed methods `dtype` is always passed.

Reply:
But to_causal_4d is part of the public api no? O this is breaking 😅